pkg/cmd/dev/testlogic.go

```diff
@@ -97,6 +97,8 @@ func (d *dev) testlogic(cmd *cobra.Command, commandLine []string) error {
 		ignoreCache = true
 	}
 
+	changes, _ := d.exec.CommandContextSilent(ctx, "git", "status")
```

Comment:
`git status` reports the changed files on top of the currently-checked-out commit. This is *not* sufficient to accomplish what we want. For example, consider that I might make a change to the logic test files, then `git commit` the result -- this wouldn't trigger since `git status` won't report any changed files, but that's incorrect. 

What we want to do is diff the changed files between the `merge-base` and `HEAD`. The function `determineAffectedTargets()` demonstrates one way to do this. You will probably have to refactor that function into a different function to do the `remote` parsing, the `merge-base` calculation, and the `diff`.

Reply:
We want to *skip code generation* if there are no updated relevant files. It has no bearing on which tests are run.